src/mono/mono/eventpipe/ep-rt-mono.c

```diff
@@ -3784,7 +3784,7 @@ get_module_event_data (
 		if (image && image->aot_module)
 			module_data->module_flags |= MODULE_FLAGS_NATIVE_MODULE;
 
-		module_data->module_il_path = image && image->filename ? image->filename : "";
+		module_data->module_il_path = image && image->filename ? image->filename : (image && image->name ? image->name : "");
```

Comment:
Just curious, in what case would we get `module_il_path = ""`?

Reply:
Hm. good point.   `image->name` is always non-null. If it's not set explicitly (for example when doing `Aseembly.Load(byte[])` it's set to `"data-%p"`.  @lateralusX is it better to have `""` or some arbitrary made-up name?

Reply:
Not sure, this value can be used by tooling to locate and potentially load the module. if we have "" I guess there is bigger chance that the tool will understand that its missing, compared to having some made up value that it will probably try to use and probably fail doing.

Reply:
This is what CoreCLR does:

```
    // if we do not have a module path yet, we put the module name
    if(bIsDynamicAssembly || ModuleILPath==NULL || wcslen(ModuleILPath) <= 2)
    {
        moduleName.SetUTF8(pModule->GetSimpleName());
        ModuleILPath = (PWCHAR)moduleName.GetUnicode();
        ModuleNativePath = (PWCHAR)pEmptyString;
    }
```

So they assume they always have a valid module name and will use that as the modules IL path.

Reply:
Ok, changed it to use the module name if available and fall back to `""`